src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -159,15 +159,21 @@ private static AnalyzerLoadFailureEventArgs CreateAnalyzerFailedArgs(Exception e
         /// </summary>
         /// <exception cref="BadImageFormatException">The PE image format is invalid.</exception>
         /// <exception cref="IOException">IO error reading the metadata.</exception>
+        [PerformanceSensitive( "https://github.com/dotnet/roslyn/issues/30449" )]
         private static ImmutableDictionary<string, ImmutableHashSet<string>> GetAnalyzerTypeNameMap(string fullPath, AttributePredicate attributePredicate)
         {
             using (var assembly = AssemblyMetadata.CreateFromFile(fullPath))
             {
+                // This is longer than strictly necessary to avoid thrashing the GC with string allocations
+                // in the call to GetFullyQualifiedTypeNames. Specifically, this checks for the presence of
+                // supported languages prior to creating the type names.
                 var typeNameMap = from module in assembly.GetModules()
                                   from typeDefHandle in module.MetadataReader.TypeDefinitions
-                                  let typeDef = module.MetadataReader.GetTypeDefinition(typeDefHandle)
-                                  let typeName = GetFullyQualifiedTypeName(typeDef, module.Module)
-                                  from supportedLanguage in GetSupportedLanguages(typeDef, module.Module, attributePredicate)
+                                  let typeDef = module.MetadataReader.GetTypeDefinition( typeDefHandle )
+                                  let supportedLanguages = GetSupportedLanguages( typeDef, module.Module, attributePredicate )
```

Comment:
Please remove the unnecessary spacing around parens.

Reply:
I mentioned this in another comment. These spacing settings are currently left to developer discretion, so on this users machine it's getting enforced in the other direction (tools Options has them as required spaces).

Reply:
Okay ... that may be true but that doesn't change what the coding style is for this repository. Spaces shouldn'd be in this position. Should be fixed before merging.

Reply:
Okay, I've addressed the issue on my end. Anything else?